File: binary_search.cpp
#include <iostream>
using namespace std;

int binarySearch(int *input, int n, int val)
{
    int start,end,mid;
    start=0;
    end=n-1;


    while(start<=end){
        mid=(start+end)/2;
        if(input[mid]==val){
            return mid;
        }
        else if(val<input[mid]){
            end = mid - 1;
        }
        else{
            start=mid+1;
        }

    }
    return -1;

}

int main()
{

	int size;
	cin >> size;
	int *input = new int[size];

	for(int i = 0; i < size; ++i)
	{
		cin >> input[i];
	}
		int val;
    cout<<"Enter the value to be searched: ";
	cin >> val;
	int index =	binarySearch(input, size, val);
    if(index == -1){
      cout<< val <<" is not present in the array";
   }
   else{
      cout<< val <<" is present at index "<< index <<" in the array";
   }

	return 0;
}//Binary Search

File: python/task/binary_search.py
#!/usr/bin/env python
# coding: utf-8

# In[ ]:


def binarySearch(nums, item):
    
    """Complete this function to perform binary search"""
    low = 0
    high = len(nums)-1
    while low<=high:
        mid = (low+high)//2
        if nums[mid]==item:
            return mid
        elif nums[mid]>item:
            high = mid-1
            continue
        else:
            low = mid+1
    return -1

def main():

    nums = [2, 4, 5, 8, 9, 20, 45, 60, 91]
    item = 60
    result = binarySearch(nums, item)

    if result != -1:
        print("FOUND! Index: {} ".format(result))
    else:
        print("NOT FOUND...")

if __name__ =="__main__":
    main()

